### DISCUSSION TITLE
How to update the entire context when a consumer modifies a value

### DISCUSSION THREAD

**Discussion by igmansilla** [lit/lit#4639] [Q&A]
Hello everyone,

I'm working on a project using Lit Element, and I'm utilizing context to share data between different components. However, I've encountered an issue when trying to update the entire context when a consumer modifies a value.

Here's my scenario: I have a provider that provides a context with multiple values. Several consumers use these values in different parts of my application. Now, when one of the consumers modifies a value in the context, I want all other consumers using that context to automatically update with the new value.

Is there any way to achieve this using Lit Element's context system? Or is there any recommended strategy for handling this type of situation?

I would greatly appreciate any guidance or suggestions you can offer.

Thank you!

**Comment by sorvell** [comment]
If I understand correctly, the issue is that mutating the contents of the context object being shared doesn't propagate that change to wherever that context object sub-property is being rendered. If you'd like to provide a [playground](https://lit.dev/playground/#sample=examples/hello-world) subset showing the problem, we can provide a specific fix that follows this general pattern...

You'll need to make sure wherever you're consuming the context data reacts to the change in the data. There are a few ways to do this and it depends on how you're consuming the data down the component tree:

1. Use an [immutable pattern](https://en.wikipedia.org/wiki/Immutable_object) to modify your context data. If it's passed to other components via Lit property bindings, then those will be triggered via this type of modification.
2. You can "consume" the data granularly where it's used and rely on the context system to propagate those granular changes.
3. You can pass the sub-property data as [signals](https://github.com/lit/lit/tree/main/packages/labs/preact-signals) and modify them as needed. If your consumers use `SignalWatcher` or `watch` then they will react to changes in the signal values.
